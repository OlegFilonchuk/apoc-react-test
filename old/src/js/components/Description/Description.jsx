import React from 'react';

import TextArea from '../TextArea/TextArea';
import withCharactersLimit from '../withCharactersLimit/withCharactersLimit';

const DescriptionComponent = withCharactersLimit(TextArea);

const Description = props => <DescriptionComponent {...props} />;

export default Description;
