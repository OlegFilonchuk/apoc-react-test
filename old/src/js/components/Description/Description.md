## Basic Description component

Simple usage of the **Desciption** component:

    const initialState = {
      value: 'Description text here'
    };

    const onChange = e => {
      setState({ value: e.target.value })
    };

    (
      <div>
        <Description onChange={onChange} value={state.value} />
      </div>
    )

Wrapped **Desciption** with static width parent and custom maxLength

    const initialState = {
      value: 'some'
    };

    const onChange = e => {
      setState({ value: e.target.value })
    };

    (
      <div style={{width: '150px'}}>
        <Description
          maxLength={16}
          onChange={onChange}
          value={state.value}
        />
      </div>
    )

**Desciption** with placeholder:

    const onChange = e => {
      setState({ value: e.target.value })
    };

    (
      <div>
        <Description
          placeholder="Enter your description here"
          onChange={onChange}
          value={state.value}
        />
      </div>
    )
