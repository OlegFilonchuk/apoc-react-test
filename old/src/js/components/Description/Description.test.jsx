import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Description from './Description';

describe('<Description>', () => {
  const mountDescription = (props, mountFn = mount) => mountFn(<Description {...props} />);

  it('should render Description component', () => {
    const props = {
      value: 'test'
    };

    const component = mountDescription(props, shallow);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should add placeholder to description', () => {
    const props = { value: 'test', placeholder: 'placeholder' };

    const component = mountDescription(props, shallow);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should run onBlur callback when Description is changed', () => {
    const onBlur = jest.fn();

    const props = {
      onBlur,
      value: 'test'
    };

    const expectedValue = 'test2';

    const component = mountDescription(props);

    component.find('textarea').simulate('blur', {
      target: {
        value: expectedValue
      }
    });

    expect(onBlur.mock.calls[0][0].target.value).toBe(expectedValue);
  });

  it('should run onChange callback when Description is changed', () => {
    const onChange = jest.fn();

    const props = {
      onChange,
      value: 'test'
    };
    const expectedValue = 'test2';

    const component = mountDescription(props);

    component.find('textarea').simulate('change', {
      target: {
        value: expectedValue
      }
    });

    expect(onChange.mock.calls[0][0].target.value).toBe(expectedValue);
  });
});
