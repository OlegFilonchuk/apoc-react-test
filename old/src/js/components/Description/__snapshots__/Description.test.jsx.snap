// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Description> should add placeholder to description 1`] = `
<TextAreaWithCharactersLimit
  className=""
  maxLength={280}
  placeholder="placeholder"
  value="test"
/>
`;

exports[`<Description> should render Description component 1`] = `
<TextAreaWithCharactersLimit
  className=""
  maxLength={280}
  value="test"
/>
`;
